### PR TITLE
[AppService]: fix #13109 - Assigning user assigned managed identity to app service slot

### LINKED ISSUE #13109: WebApp:Assigning user assigned managed identity to app service slot
## Description of the new feature
For some reason, the identity of a web app deployment slot is not the same as the identity of the original app service. Given this, azure powershell has no method of assigning a user defined managed identity for a deployment slot. So if I have an app service that uses user defined managed identity, and use `new-azwebappslot` to create a new slot, I have no way of assigning the user defined managed identity to it using azure powershell. Currently I have to resort to an ARM template to assign the identity later.

There are several improvements possible:

- an option in `new-azwebappslot` to also copy the identity settings
- an option in `Set-AzWebAppSlot` to set the user defined managed identity, so I can set it after creating the slot
- an option in `Set-AzWebApp` to set the user defined managed identity, which also seems missing

To be able to not having to resort to an ARM template, I need either option 1 or 2. But I prefer all of them to be implemented

## Proposed implementation details (optional)
In `new-azwebappslot` it can be a switch: `-copyidentity`
The other methods are a bit more complicated since you can add multiple user defined managed identities to an app service

### PR DESCRIPTION
**Pull request by Kotasudhakarreddy**
#13109 
https://github.com/Azure/azure-powershell/issues/13637

## Description
For some reason, the identity of a web app deployment slot is not the same as the identity of the original app service. Given this, azure powershell has no method of assigning a user defined managed identity for a deployment slot. So if I have an app service that uses user defined managed identity, and use new-azwebappslot to create a new slot, I have no way of assigning the user defined managed identity to it using azure powershell. Currently I have to resort to an ARM template to assign the identity later.

There are several improvements possible and the best used is

an option in new-azwebappslot to also copy the identity settings

## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Review by BethanyZhou — DISMISSED** [review]

File: src/Websites/Websites/Cmdlets/DeploymentSlots/NewAzureWebAppSlot.cs [hunk 1]
```diff
@@ -167,8 +170,8 @@ private void UpdateConfigIfNeeded(Site site)
 		public override void ExecuteCmdlet()
         {
             base.ExecuteCmdlet();
-            
-            CloningInfo cloningInfo = null;
+			ManagedServiceIdentity sourceIdentity = null;
+			CloningInfo cloningInfo = null;
```
Comment (BethanyZhou) [inline src/Websites/Websites/Cmdlets/DeploymentSlots/NewAzureWebAppSlot.cs:174]:
```suggestion
            ManagedServiceIdentity sourceIdentity = null;
            CloningInfo cloningInfo = null;
```
Correct indents please

**Review by BethanyZhou — CHANGES_REQUESTED** [review]

File: src/Websites/Websites/Cmdlets/DeploymentSlots/NewAzureWebAppSlot.cs (hunk 1 above)
Comment (Kotasudhakarreddy) [reply src/Websites/Websites/Cmdlets/DeploymentSlots/NewAzureWebAppSlot.cs]: Done

**Comment by BethanyZhou** [comment]
Please approve the PR I raised to fix indent issues in your branch first. https://github.com/Azure/azure-powershell/pull/20985/files

GitHub does differentiate space and tab, I'd suggest using VSCode to detect their differences.

**Review by BethanyZhou — APPROVED** [review]
LGTM

**Pull request merged**